src/ctx.rs

```diff
@@ -14,27 +14,48 @@ pub(crate) fn get_app_config() -> &'static AppConfig {
     })
 }
 
+/// Configuration settings for the application, loaded typically from a JSON configuration file.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub(crate) struct AppConfig {
+    /// Optional server port to listen on. If None in config file, then 5000 is default.
     pub(crate) port: Option<u16>,
+    /// Redis database connection string.
     pub(crate) redis_connection_string: String,
+    /// File path to the public key used for cryptographic operations.
     pub(crate) pubkey_path: String,
+    /// File path to the private key used for cryptographic operations.
     pub(crate) privkey_path: String,
+    /// Optional token expiration time in seconds. If None then 3600 is default.
     pub(crate) token_expiration_time: Option<i64>,
+    /// Routing configurations for proxying requests.
     pub(crate) proxy_routes: Vec<ProxyRoute>,
+    /// Default rate limiting settings for request handling.
     pub(crate) rate_limiter: RateLimiter,
 }
 
+/// Defines a routing rule for proxying requests from an inbound route to an outbound URL
+/// based on a specified proxy type and additional authorization and method filtering criteria.
 #[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
 pub(crate) struct ProxyRoute {
+    /// The incoming route pattern.
     pub(crate) inbound_route: String,
+    /// The target URL to which requests are forwarded.
     pub(crate) outbound_route: String,
+    /// The type of proxying to perform (e.g., JSON-RPC Call, HTTP GET).
+    pub(crate) proxy_type: ProxyType,
+    /// Whether authorization is required for this route.
     #[serde(default)]
     pub(crate) authorized: bool,
+    // Specific HTTP methods allowed for this route.
```

Comment:
This field isn't about HTTP methods, it's for the application/rpc methods from quicknode eth nodes. Maybe rename the field into `allowed_rpc_methods` to avoid this confusion.